src/Microsoft.Management.Configuration.Processor/Public/ConfigurationSetProcessorFactory.cs

```diff
@@ -29,27 +30,66 @@ public ConfigurationSetProcessorFactory(ConfigurationProcessorType type, IConfig
             this.properties = properties;
         }
 
+        /// <summary>
+        /// Diagnostics event; useful for logging and/or verbose output.
+        /// </summary>
+        public event EventHandler<DiagnosticInformation>? Diagnostics;
+
+        /// <summary>
+        /// Gets or sets the minimum diagnostic level to send.
+        /// </summary>
+        public DiagnosticLevel MinimumLevel { get; set; } = DiagnosticLevel.Informational;
+
         /// <summary>
         /// Gets the configuration unit processor details for the given unit.
         /// </summary>
         /// <param name="set">Configuration Set.</param>
         /// <returns>Configuration set processor.</returns>
         public IConfigurationSetProcessor CreateSetProcessor(ConfigurationSet set)
         {
-            var envFactory = new ProcessorEnvironmentFactory(this.type);
-            var processorEnvironment = envFactory.CreateEnvironment();
-            processorEnvironment.ValidateRunspace();
-
-            if (this.properties is not null)
+            try
             {
-                var additionalPsModulePaths = this.properties.AdditionalModulePaths;
-                if (additionalPsModulePaths is not null)
+                this.OnDiagnostics(DiagnosticLevel.Verbose, $"Creating set processor for `{set.Name}`...");
+
+                var envFactory = new ProcessorEnvironmentFactory(this.type);
+                var processorEnvironment = envFactory.CreateEnvironment();
+                processorEnvironment.ValidateRunspace();
+
+                if (this.properties is not null)
                 {
-                    processorEnvironment.PrependPSModulePaths(additionalPsModulePaths);
+                    var additionalPsModulePaths = this.properties.AdditionalModulePaths;
+                    if (additionalPsModulePaths is not null)
+                    {
+                        processorEnvironment.PrependPSModulePaths(additionalPsModulePaths);
+                    }
                 }
+
+                this.OnDiagnostics(DiagnosticLevel.Verbose, "... done creating set processor.");
+
+                return new ConfigurationSetProcessor(processorEnvironment, set) { SetProcessorFactory = this };
+            }
+            catch (Exception ex)
+            {
+                this.OnDiagnostics(DiagnosticLevel.Error, ex.ToString());

```

Comment:
Do you know how much ToString outputs for a RuntimeException? If it doesn't include much from the ErrorRecord we should also log that. I found ScriptStackTrace to be really helpful.

https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.errorrecord?view=powershellsdk-7.3.0#properties